src/Components/TodoList.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from "react";
+import "./Todo.scss";
+import logo from "../assests/images/logo.jpg";
+import TodoItem from "./TodoItem";
+import { Priority, Todo } from "../types";
+
+const TodoList: React.FC = () => {
+  const [task, setTask] = useState("");
+  const [priority, setPriority] = useState<Priority>("Low");
+  const [fillter, setFillter] = useState<Priority | "All">("All");
+  const [tasks, setTasks] = useState<Todo[]>(() => {
+    const savedTasksInLocalStorge = localStorage.getItem("task");
+    return savedTasksInLocalStorge ? JSON.parse(savedTasksInLocalStorge) : [];
+  });
+
+  useEffect(() => {
+    localStorage.setItem("task", JSON.stringify(tasks));
+  }, [tasks]);
+
+  const removeTask = (index: number) => {
+    setTasks(tasks.filter((_, i) => i !== index));
+  };
+
+  const filterTasks = tasks.filter((task) => {
+    if (fillter === "All") return true;
+    return task.priority === fillter;
+  });
```

Comment:
Dont store the tasks again into `filterTasks`, so do the filter directly once render the ui